paimon-spark/paimon-spark-common/src/test/scala/org/apache/paimon/spark/sql/DataFrameWriteTest.scala

```diff
@@ -182,92 +183,120 @@ class DataFrameWriteTest extends PaimonSparkTestBase {
-
-            spark.sql(s"""
-                         |CREATE TABLE T (a INT, b STRING)
-                         |TBLPROPERTIES ($prop)
-                         |""".stripMargin)
-
-            val paimonTable = loadTable("T")
-            val location = paimonTable.location().toString
-
-            val df1 = Seq((1, "a"), (2, "b")).toDF("a", "b")
-            df1.write.format("paimon").mode("append").save(location)
-            checkAnswer(
-              spark.sql("SELECT * FROM T ORDER BY a, b"),
-              Row(1, "a") :: Row(2, "b") :: Nil)
-
-            // Case 1: two additional fields
-            val df2 = Seq((1, "a2", 123L, Map("k" -> 11.1)), (3, "c", 345L, Map("k" -> 33.3)))
-              .toDF("a", "b", "c", "d")
-            df2.write
-              .format("paimon")
-              .mode("append")
-              .option("write.merge-schema", "true")
-              .save(location)
-            val expected2 = if (hasPk) {
-              Row(1, "a2", 123L, Map("k" -> 11.1)) ::
-                Row(2, "b", null, null) :: Row(3, "c", 345L, Map("k" -> 33.3)) :: Nil
-            } else {
-              Row(1, "a", null, null) :: Row(1, "a2", 123L, Map("k" -> 11.1)) :: Row(
-                2,
-                "b",
-                null,
-                null) :: Row(3, "c", 345L, Map("k" -> 33.3)) :: Nil
-            }
-            checkAnswer(spark.sql("SELECT * FROM T ORDER BY a, b"), expected2)
-
-            // Case 2: two fields with the evolved types: Int -> Long, Long -> Decimal
-            val df3 = Seq(
-              (2L, "b2", BigDecimal.decimal(234), Map("k" -> 22.2)),
-              (4L, "d", BigDecimal.decimal(456), Map("k" -> 44.4))).toDF("a", "b", "c", "d")
-            df3.write
-              .format("paimon")
-              .mode("append")
-              .option("write.merge-schema", "true")
-              .save(location)
-            val expected3 = if (hasPk) {
-              Row(1L, "a2", BigDecimal.decimal(123), Map("k" -> 11.1)) :: Row(
-                2L,
-                "b2",
-                BigDecimal.decimal(234),
-                Map("k" -> 22.2)) :: Row(3L, "c", BigDecimal.decimal(345), Map("k" -> 33.3)) :: Row(
-                4L,
-                "d",
-                BigDecimal.decimal(456),
-                Map("k" -> 44.4)) :: Nil
-            } else {
-              Row(1L, "a", null, null) :: Row(
-                1L,
-                "a2",
-                BigDecimal.decimal(123),
-                Map("k" -> 11.1)) :: Row(2L, "b", null, null) :: Row(
-                2L,
-                "b2",
-                BigDecimal.decimal(234),
-                Map("k" -> 22.2)) :: Row(3L, "c", BigDecimal.decimal(345), Map("k" -> 33.3)) :: Row(
-                4L,
-                "d",
-                BigDecimal.decimal(456),
-                Map("k" -> 44.4)) :: Nil
-            }
-            checkAnswer(spark.sql("SELECT * FROM T ORDER BY a, b"), expected3)
-
+  fileFormats.foreach {
```

Comment:
only changed: `fileFormats.foreach` and `// Case 3: insert Decimal(20,18) to Decimal(38,18)`